### PR TITLE
Shared github workflows

### PR DESCRIPTION
**Pull request by dprince**
This PR switches the operator to use shared build workflows for building operator, bundle, and catalog/index images.

Additionally fixes the opm image tag to one that exists.

Jira: [OSP-29146](https://issues.redhat.com//browse/OSP-29146)

### CODE REVIEW

**Comment by dprince** [comment]
recheck

**Comment by fao89** [comment]
recheck

File: Makefile [hunk 1]
```diff
@@ -266,7 +266,7 @@ ifeq (,$(shell which opm 2>/dev/null))
 	set -e ;\
 	mkdir -p $(dir $(OPM)) ;\
 	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
-	curl -sSLo $(OPM) https://github.com/operator-framework/operator-registry/releases/download/v1.31.0/$${OS}-$${ARCH}-opm ;\
+	curl -sSLo $(OPM) https://github.com/operator-framework/operator-registry/releases/download/v1.29.0/$${OS}-$${ARCH}-opm ;\
```
Comment (jpodivin) [inline Makefile:269]: Why the curl downgrade? The reasoning should be in the commit.

**Review by jpodivin — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (slagle) [reply Makefile:269]: It is I believe.

```
Additionally fixes the opm image tag to one that exists.
```

**Review by slagle — COMMENTED** [review]
+2

File: Makefile (hunk 1 above)
Comment (slagle) [reply Makefile:269]: @jpodivin WDYT? Can we move forward with merging this one?

Comment (jpodivin) [reply Makefile:269]: I suppose that makes sense. It may have been better as a separate patch but if that's not possible, why not.

**Review by jpodivin — APPROVED** [review]

**Pull request merged**